docs/draft-alakuijala-brotli-07.nroff

```diff
@@ -1008,9 +1008,10 @@ lengths and distances, respectively, is encoded in the meta-block
 header, and it must equal to the largest block type plus one in that
 block category. In other words, the set of literal, insert-and-copy
 length and distance block types must be [0..NBLTYPESL-1],
-[0..NBLTYPESI-1], and [0..NBLTYPESD-1], respectively. From this it
-follows that the alphabet size of literal, insert-and-copy length and
-distance block type codes is NBLTYPES + 2, NBLTYPESI + 2 and
+[0..NBLTYPESI-1], and [0..NBLTYPESD-1], respectively. If a block type
+has a value outside its respective range, the stream should be rejected as
+invalid. From this it follows that the alphabet size of literal, insert-and-copy
+length and distance block type codes is NBLTYPES + 2, NBLTYPESI + 2 and
```

Comment:
This was probably not very clear. Hopefully better explanation: In my implementation, I was under the impression that a simple prefix code can define arbitrary symbols, as long as they are encoded with ALPHABET_SIZE bits. However, the spec clearly mentions that the integer value of a symbol must be within [0..ALPHABET_SIZE-1].

I had missed that before, thus my implementation was wrong. That allowed a corrupt stream to inject invalid block types in a prefix code. I _thought_ the underlying cause was in not rejecting invalid block types, but the real cause was in not rejecting invalid symbols in simple prefix codes.